include/drivers/hwinfo.h

```diff
@@ -54,6 +66,30 @@ __syscall ssize_t hwinfo_get_device_id(uint8_t *buffer, size_t length);
 
 ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length);
 
+/**
+ * @brief      Retrieve cause of device reset.
+ *
+ * @param      cause  OR'd flags TBD
+ *
```

Comment:
Add a warning that states that on some MCUs the mere fact or reading the flags will also clear them, which is a problematic side-effect.

Reply:
Alternatively, implementations that suffer from this limitation could cache the value of the reset reason so that the value can be read using this call multiple times. This wouldn't work with a bootloader, but it would work if this API is called multiple times inside the main image.